packages/acceptance-tests/pkg-tests-specs/sources/commands/pack.test.js

```diff
@@ -287,5 +288,67 @@ describe(`Commands`, () => {
         await expect(stdout).toMatch(/package\.json/);
       }),
     );
+
+    test(
+      `the filename is non-descriptive by default`,
+      makeTemporaryEnv({
+        name: '@berry/core',
+        version: '0.0.1',
+      }, async ({path, run, source}) => {
+        await run(`install`);
+
+        const {stdout} = await run(`pack`);
+        await expect(stdout).toMatch(/package\.json/);
+        await expect(stdout).toMatch(/Package archive generated in .+?\/package\.tgz/);
+      }),
+    );
+
+    test(
+      `can generate an archive with a descriptive name`,
+      makeTemporaryEnv({
+        name: '@berry/core',
+        version: '0.0.1',
+      }, async ({path, run, source}) => {
+        await run(`install`);
+
+        const {stdout} = await run(`pack`, `--out`, `./%n.tgz`);
+        await expect(stdout).toMatch(/package\.json/);
+        await expect(stdout).toMatch(/Package archive generated in .+?\/berry-core-0\.0\.1\.tgz/);
+      }),
+    );
+
+    test(
+      `the descriptive name doesn't add empty slug parts if the package has no scope`,
+      makeTemporaryEnv({
+        name: 'core',
+        version: '0.0.1',
+      }, async ({path, run, source}) => {
+        await run(`install`);
+
+        const {stdout} = await run(`pack`, `--out`, `./%n.tgz`);
+        await expect(stdout).toMatch(/package\.json/);
+        await expect(stdout).toMatch(/Package archive generated in .+?\/core-0\.0\.1\.tgz/);
+      }),
+    );
+
+    test(
+      `can output the archive in a absolute destination`,
+      makeTemporaryEnv({
+        name: '@berry/core',
+        version: '0.0.1',
+      }, async ({path, run, source}) => {
+        await run(`install`);
+
+        let stderr;
+        try {
+          ({stderr} = await run(`pack`, `--out`, `/artifacts/berry-core.tgz`));
+        } catch (error) {
+          ({stderr} = error);
+        }
+
+        const expectedPath = npath.resolve('/artifacts/berry-core.tgz');
```

Comment:
`npath` works with the native paths (it's basically an alias of `require('path')`), so this doesn't work on Windows (you'd have to do something like `npath.resolve(fromPortablePath('...'))`).

Regardless, why not use `${path}/berry-core.tgz` as absolute path? This way you can check whether the file exists, not whether the write failed.

Reply:
I wanted to test that we can use paths outside of the cwd. Using `path` wouldn't actually test that, no?

Reply:
In this case I'd suggest to use a temporary directory (`xfs.mktempPromise`). This deals with cross-platform issues.